### PR TITLE
Add back button to HCD guide page

### PR DESCRIPTION
**Pull request by clmedders**
## Summary

This Pr will update a bug/nice to have for the HCD page and guides.. it was expressed that a "go back" button was needed to allow users to go back to the landing page from any part of the guide. 

### Preview

[Link to Preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/cm-hcd-back-button/guides/hcd/)

### Solution
Update the p tag with the kicker class, turn it into a link element, and add the uswds `class="usa-link"` to keep it line with USWDS. 

The reason for changing the p tag into a link tag was to indicate to the user that the string is a link and the text inside gives context that it will take them back to the landing page for the HCD guides. 

### How To Test

1. Visit [Preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/cm-hcd-back-button/guides/hcd/?dg)
2. click though the guides and check on the link `HCD GUIDE SERIES` and verify that you are taken back to the landing page for HCD (see screenshot below for reference)
3. remove the `hcd` and replace with `dap` and make sure that the kicker on that page is not linked and is just a string.

Screenshot 2023-10-05 at 11 13 14 AM

---

### Dev Checklist

- [x] PR has correct labels
- [x] A11y testing (voice over testing, meets WCAG, run axe tools)
- [x] Code is formatted properly

### CODE REVIEW

**Comment by ToniBonittoGSA** [comment]
@clmedders 

I'm getting a 404 when I click the kicker link in the preview

Also, pls remove the `?dg` from the code.

**Review by mejiaj — CHANGES_REQUESTED** [review]
I like the idea of linking the `kicker` to avoid additional cruft on the page.

Unfortunately, hardcoding it will have a negative effect on other guides  ― like [DAP Guide](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/cm-hcd-back-button/guides/dap/add-your-site-dap/). It links to the HCD landing.

 We should default to the parent link otherwise link to grandparent if page is a node via `.IsNode`.

File: themes/digital.gov/layouts/partials/core/guides/guide-header.html [hunk 1]
```diff
@@ -6,11 +6,20 @@
...
 {{- $glossary := cond .IsNode .Params.glossary .Parent.Params.glossary -}}
+{{ $landingURL := $guideData.landingURL }}
 
 
 <header class="dg-guide__header grid-container grid-container-desktop">
   <div class="dg-guide__header-title">
-    <p class="kicker">{{- $kicker | markdownify -}}</p>
+    {{ if isset $guideData "landingURL" }}
+      <a class="usa-link kicker" href="{{ $landingURL | relURL }}"
+        >{{- $kicker | markdownify -}}</a
```
[the remark this answers is not in the document]
Comment (clmedders) [reply themes/digital.gov/layouts/partials/core/guides/guide-header.html:16]: thats a good question... i just asked to find out what it does but makes sense if you are going to repeat it to to have it simple

**Review by mejiaj — APPROVED** [review]
LGTM, I tested the link on HCD & DAP.

**Pull request merged**